x-pack/plugins/infra/server/lib/alerting/metric_threshold/lib/evaluate_alert.ts

```diff
@@ -121,7 +122,10 @@ const getMetric: (
   const intervalAsSeconds = getIntervalInSeconds(interval);
   const intervalAsMS = intervalAsSeconds * 1000;
 
-  const to = roundTimestamp(timeframe ? timeframe.end : Date.now(), timeUnit);
+  const to = moment(timeframe ? timeframe.end : Date.now())
```

Comment:
Looks like we're still using this `roundTimestamp` util for calculating the `from` value, but calculating it 'by hand' here for the `to`, is that right?

Reply:
Yeah the problem is that we don't actually want to round the time since that function could round down. Doing it by hand forces it to round up always and get that last bucket.